src/Network/Pusher/Data.hs

```diff
@@ -59,26 +60,54 @@ type AppSecret = B.ByteString
 
 -- |All the required configuration needed to interact with the API.
 data Pusher = Pusher
-  { pusherHost :: T.Text
-  , pusherPath :: T.Text
-  , pusherCredentials :: Credentials
+  { pusherHost              :: T.Text
+  , pusherPath              :: T.Text
+  , pusherCredentials       :: Credentials
   , pusherConnectionManager :: Manager
   }
 
 -- |The credentials for the current app.
 data Credentials = Credentials
-  { credentialsAppID :: AppID
-  , credentialsAppKey :: AppKey
+  { credentialsAppID     :: AppID
+  , credentialsAppKey    :: AppKey
   , credentialsAppSecret :: AppSecret
+  , credentialsCluster   :: Maybe Cluster
   }
 
 instance A.FromJSON Credentials where
   parseJSON (A.Object v) = Credentials
     <$> v .: "app-id"
     <*> (encodeUtf8 <$> v .: "app-key")
     <*> (encodeUtf8 <$> v .: "app-secret")
+    <*> v.:? "app-cluster"
   parseJSON v2 = failExpectObj v2
 
+-- |The cluster the current app resides on.
+data Cluster
+  = Mt1 -- ^ us-east-1
+  | Eu  -- ^ eu-west-1
+  | Ap1 -- ^ ap-southeast-1 (Singapore)
+  | Ap2 -- ^ ap-south-1 (Mumbai)
+
+-- The possible cluster suffix given in a host name
+renderClusterSuffix :: Cluster -> T.Text
+renderClusterSuffix cluster = case cluster of
+  Mt1 -> ""
```

Comment:
Btw, this edge case is not strictly necessary as the routes also exist for `-mt1`.